Libraries/Opc.Ua.Client.ComplexTypes/ComplexTypeSystem.cs

```diff
@@ -1094,11 +1095,11 @@ ExpandedNodeId xmlEncodingId
                 {
                     if (field.ValueRank < 0) // scalar
                     {
-                        fieldBuilder.AddField(field, (fieldBuilder as ComplexTypeFieldBuilder).StructureTypeBuilder, order);
+                        fieldBuilder.AddField(field, fieldBuilder.StructureTypeBuilder, order);
```

Comment:
Looks like this breaks encapsulation of IComplexTypeFactory in the current implementation, so we need a way to pass the parameter without relying on the TypeBuilder implementation.

Reply:
as an alternative we could add this property to IComplexTypeFieldBuilder:

Type GetStructureType(int valueRank)